Index: ctlPuttyPanel.cs
===================================================================
--- ctlPuttyPanel.cs	(revision 117)
+++ ctlPuttyPanel.cs	(working copy)
@@ -1,4 +1,4 @@
-﻿/*
+/*
  * Copyright (c) 2009 Jim Radford http://www.jimradford.com
  * Copyright (c) 2012 John Peterson
  * Permission is hereby granted, free of charge, to any person obtaining a copy
@@ -67,6 +67,7 @@
                 args += (!String.IsNullOrEmpty(m_Session.Password) && m_Session.Password.Length > 0) ? "-pw " + m_Session.Password + " " : "";
                 args += "-P " + m_Session.Port + " ";
                 args += (!String.IsNullOrEmpty(m_Session.PuttySession)) ? "-load \"" + m_Session.PuttySession + "\" " : "";
+                args += (!String.IsNullOrEmpty(m_Session.ExtraArgs) ? m_Session.ExtraArgs + " " : "");
                 args += (!String.IsNullOrEmpty(m_Session.Username) && m_Session.Username.Length > 0) ? m_Session.Username + "@" : "";
                 args += m_Session.Host;
             } 
Index: Data/SessionData.cs
===================================================================
--- Data/SessionData.cs	(revision 117)
+++ Data/SessionData.cs	(working copy)
@@ -1,4 +1,4 @@
-﻿/*
+/*
  * Copyright (c) 2009 Jim Radford http://www.jimradford.com
  * Permission is hereby granted, free of charge, to any person obtaining a copy
  * of this software and associated documentation files (the "Software"), to deal
@@ -135,6 +135,14 @@
             set { _Password = value; }
         }
 
+        private string _ExtraArgs;
+        [XmlAttribute]
+        public string ExtraArgs
+        {
+            get { return _ExtraArgs; }
+            set { _ExtraArgs = value; }
+        }
+
         /* Unused...ignore for now
         private string _LastPath = ".";
         public string LastPath
Index: dlgEditSession.cs
===================================================================
--- dlgEditSession.cs	(revision 117)
+++ dlgEditSession.cs	(working copy)
@@ -1,4 +1,4 @@
-﻿/*
+/*
  * Copyright (c) 2009 Jim Radford http://www.jimradford.com
  * Permission is hereby granted, free of charge, to any person obtaining a copy
  * of this software and associated documentation files (the "Software"), to deal
@@ -54,6 +54,7 @@
                 this.textBoxSessionName.Text = Session.SessionName;
                 this.textBoxHostname.Text = Session.Host;
                 this.textBoxPort.Text = Session.Port.ToString();
+                this.textBoxExtraArgs.Text = Session.ExtraArgs;
                 this.textBoxUsername.Text = Session.Username;
 
                 switch (Session.Proto)
@@ -140,6 +141,7 @@
             Session.SessionName = textBoxSessionName.Text.Trim();
             Session.PuttySession = comboBoxPuttyProfile.Text.Trim();
             Session.Host = textBoxHostname.Text.Trim();
+            Session.ExtraArgs = textBoxExtraArgs.Text.Trim();
             Session.Port = int.Parse(textBoxPort.Text.Trim());
             Session.Username = textBoxUsername.Text.Trim();
             Session.SessionId = SessionData.CombineSessionIds(SessionData.GetSessionParentId(Session.SessionId), Session.SessionName);
@@ -166,6 +168,7 @@
             string host = this.textBoxHostname.Text;
             bool isCygterm = this.radioButtonCygterm.Checked;
             this.textBoxPort.Enabled = !isCygterm;
+            this.textBoxExtraArgs.Enabled = !isCygterm;
             this.textBoxUsername.Enabled = !isCygterm;
 
             if (isCygterm)
Index: dlgEditSession.Designer.cs
===================================================================
--- dlgEditSession.Designer.cs	(revision 117)
+++ dlgEditSession.Designer.cs	(working copy)
@@ -1,4 +1,4 @@
-﻿namespace SuperPutty
+namespace SuperPutty
 {
     partial class dlgEditSession
     {
@@ -42,6 +42,8 @@
             this.radioButtonRlogin = new System.Windows.Forms.RadioButton();
             this.radioButtonTelnet = new System.Windows.Forms.RadioButton();
             this.radioButtonRaw = new System.Windows.Forms.RadioButton();
+            this.textBoxExtraArgs = new System.Windows.Forms.TextBox();
+            this.label6 = new System.Windows.Forms.Label();
             this.buttonSave = new System.Windows.Forms.Button();
             this.buttonCancel = new System.Windows.Forms.Button();
             this.label4 = new System.Windows.Forms.Label();
@@ -214,14 +216,31 @@
             this.radioButtonRaw.UseVisualStyleBackColor = true;
             this.radioButtonRaw.CheckedChanged += new System.EventHandler(this.radioButtonRaw_CheckedChanged);
             // 
+            // label6
+            // 
+            this.label6.AutoSize = true;
+            this.label6.Font = new System.Drawing.Font("Segoe UI", 9F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
+            this.label6.Location = new System.Drawing.Point(12, 224);
+            this.label6.Name = "label6";
+            this.label6.Size = new System.Drawing.Size(132, 15);
+            this.label6.TabIndex = 14;
+            this.label6.Text = "Extra PuTTY Arguments";
+            // 
+            // textBoxExtraArgs
+            // 
+            this.textBoxExtraArgs.Location = new System.Drawing.Point(150, 222);
+            this.textBoxExtraArgs.Name = "textBoxExtraArgs";
+            this.textBoxExtraArgs.Size = new System.Drawing.Size(249, 20);
+            this.textBoxExtraArgs.TabIndex = 6;
+            // 
             // buttonSave
             // 
             this.buttonSave.Anchor = ((System.Windows.Forms.AnchorStyles)((System.Windows.Forms.AnchorStyles.Bottom | System.Windows.Forms.AnchorStyles.Right)));
             this.buttonSave.Enabled = false;
-            this.buttonSave.Location = new System.Drawing.Point(246, 234);
+            this.buttonSave.Location = new System.Drawing.Point(246, 257);
             this.buttonSave.Name = "buttonSave";
             this.buttonSave.Size = new System.Drawing.Size(75, 23);
-            this.buttonSave.TabIndex = 6;
+            this.buttonSave.TabIndex = 7;
             this.buttonSave.Text = "Save";
             this.buttonSave.UseVisualStyleBackColor = true;
             this.buttonSave.Click += new System.EventHandler(this.buttonSave_Click);
@@ -230,7 +249,7 @@
             // 
             this.buttonCancel.Anchor = ((System.Windows.Forms.AnchorStyles)((System.Windows.Forms.AnchorStyles.Bottom | System.Windows.Forms.AnchorStyles.Right)));
             this.buttonCancel.DialogResult = System.Windows.Forms.DialogResult.Cancel;
-            this.buttonCancel.Location = new System.Drawing.Point(327, 234);
+            this.buttonCancel.Location = new System.Drawing.Point(327, 257);
             this.buttonCancel.Name = "buttonCancel";
             this.buttonCancel.Size = new System.Drawing.Size(75, 23);
             this.buttonCancel.TabIndex = 7;
@@ -244,18 +263,18 @@
             this.label4.Font = new System.Drawing.Font("Segoe UI", 9F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
             this.label4.Location = new System.Drawing.Point(12, 172);
             this.label4.Name = "label4";
-            this.label4.Size = new System.Drawing.Size(114, 15);
+            this.label4.Size = new System.Drawing.Size(121, 15);
             this.label4.TabIndex = 9;
-            this.label4.Text = "Putty Session Profile";
+            this.label4.Text = "PuTTY Session Profile";
             // 
             // comboBoxPuttyProfile
             // 
             this.comboBoxPuttyProfile.Anchor = ((System.Windows.Forms.AnchorStyles)(((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Left)
                         | System.Windows.Forms.AnchorStyles.Right)));
             this.comboBoxPuttyProfile.FormattingEnabled = true;
-            this.comboBoxPuttyProfile.Location = new System.Drawing.Point(127, 170);
+            this.comboBoxPuttyProfile.Location = new System.Drawing.Point(150, 170);
             this.comboBoxPuttyProfile.Name = "comboBoxPuttyProfile";
-            this.comboBoxPuttyProfile.Size = new System.Drawing.Size(272, 21);
+            this.comboBoxPuttyProfile.Size = new System.Drawing.Size(249, 21);
             this.comboBoxPuttyProfile.TabIndex = 4;
             this.comboBoxPuttyProfile.Text = "Default Settings";
             this.comboBoxPuttyProfile.TextChanged += new System.EventHandler(this.sessionForm_TextChanged);
@@ -280,7 +299,7 @@
             // 
             this.label5.AutoSize = true;
             this.label5.Font = new System.Drawing.Font("Segoe UI", 9F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
-            this.label5.Location = new System.Drawing.Point(30, 199);
+            this.label5.Location = new System.Drawing.Point(12, 198);
             this.label5.Name = "label5";
             this.label5.Size = new System.Drawing.Size(93, 15);
             this.label5.TabIndex = 12;
@@ -290,9 +309,9 @@
             // 
             this.textBoxUsername.Anchor = ((System.Windows.Forms.AnchorStyles)(((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Left)
                         | System.Windows.Forms.AnchorStyles.Right)));
-            this.textBoxUsername.Location = new System.Drawing.Point(129, 197);
+            this.textBoxUsername.Location = new System.Drawing.Point(150, 196);
             this.textBoxUsername.Name = "textBoxUsername";
-            this.textBoxUsername.Size = new System.Drawing.Size(270, 20);
+            this.textBoxUsername.Size = new System.Drawing.Size(249, 20);
             this.textBoxUsername.TabIndex = 5;
             // 
             // errorProvider
@@ -306,7 +325,9 @@
             this.AutoScaleDimensions = new System.Drawing.SizeF(6F, 13F);
             this.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font;
             this.CancelButton = this.buttonCancel;
-            this.ClientSize = new System.Drawing.Size(411, 269);
+            this.ClientSize = new System.Drawing.Size(411, 292);
+            this.Controls.Add(this.label6);
+            this.Controls.Add(this.textBoxExtraArgs);
             this.Controls.Add(this.textBoxUsername);
             this.Controls.Add(this.label5);
             this.Controls.Add(this.groupBox2);
@@ -350,5 +371,7 @@
         private System.Windows.Forms.TextBox textBoxUsername;
         private System.Windows.Forms.RadioButton radioButtonCygterm;
         private System.Windows.Forms.ErrorProvider errorProvider;
+        private System.Windows.Forms.Label label6;
+        private System.Windows.Forms.TextBox textBoxExtraArgs;
     }
 }
\ No newline at end of file
